com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -8,101 +8,362 @@
+        public LoadSceneMode LoadSceneMode;
 
         /// <summary>
-        /// Delegate for when all clients have reported to the server that they have completed scene transition or timed out
-        /// <see cref='OnNotifyServerAllClientsLoadedScene'/>
+        /// Excluding <see cref="SceneEventData.SceneEventTypes.S2C_Event_Sync"/> and <see cref="SceneEventData.SceneEventTypes.C2S_Event_Sync_Complete"/>
+        /// This will be set to the scene name that the event pertains to.
         /// </summary>
-        public delegate void NotifyServerAllClientsLoadedSceneDelegate(SceneSwitchProgress progress, bool timedOut);
+        public string SceneName;
 
         /// <summary>
-        /// Delegate for when the clients get notified by the server that all clients have completed their scene transitions.
-        /// <see cref='OnNotifyClientAllClientsLoadedScene'/>
+        /// When a scene is loaded, the Scene structure is returned.
         /// </summary>
-        public delegate void NotifyClientAllClientsLoadedSceneDelegate(ulong[] clientIds, ulong[] timedOutClientIds);
+        public Scene Scene;
 
         /// <summary>
-        /// Event that is invoked when the scene is switched
+        /// Events that always set <see cref="ClientId"/> to the local client identifier
+        /// and only triggered locally:
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_Load"/>
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_Unload"/>
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_Sync"/>
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_ReSync"/>
+        ///
+        /// Events that always set <see cref="ClientId"/> to the local client identifier,
+        /// are triggered locally, and a host or server will trigger externally generated
+        /// scene event message types (i.e. sent by a client):
+        /// <see cref="SceneEventData.SceneEventTypes.C2S_UnloadComplete"/>
+        /// <see cref="SceneEventData.SceneEventTypes.C2S_LoadComplete"/>
+        /// <see cref="SceneEventData.SceneEventTypes.C2S_SyncComplete"/>
+        ///
+        /// Events that always set <see cref="ClientId"/> to the ServerId:
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_LoadComplete"/>
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_UnLoadComplete"/>
         /// </summary>
-        public event SceneSwitchedDelegate OnSceneSwitched;
+        public ulong ClientId;
 
         /// <summary>
-        /// Event that is invoked when a local scene switch has started
+        /// List of clients that completed a loading or unloading event
+        /// Applies only to:
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_LoadComplete"/>
+        /// <see cref="SceneEventData.SceneEventTypes.S2C_UnLoadComplete"/>
         /// </summary>
-        public event SceneSwitchStartedDelegate OnSceneSwitchStarted;
+        public List<ulong> ClientsThatCompleted;
```

Comment:
Since a client can only occur once in this container, what about a set?

Reply:
I guess we could do that.  It really was for the simplicity of converting to an Array.  So, at the time I was thinking more along the lines of "serializing and deserializing" from SceneEventData perspective.  Since I used lists in SceneEventData it made sense to just use a list here without having any funky conversion between SceneEventData and SceneEvent.

While searching for a specific ulong value could possibly be slightly faster with a HashSet than a List, I don't think we will have enough user identifiers to make a noticeable difference with this specific use case scenario?  Other than possible performance gain when searching, what other benefits we would gain by doing that?  
_(if there is a compelling case then I am most definitely up to make that change)_